frontend/model/contracts/group.js

```diff
@@ -747,22 +806,34 @@ sbp('chelonia/defineContract', {
       // They MUST NOT call 'commit'!
       // They should only coordinate the actions of outside contracts.
       // Otherwise `latestContractState` and `handleEvent` will not produce same state!
-      async sideEffect ({ meta }, { state }) {
-        const rootState = sbp('state/vuex/state')
-        // TODO: per #257 this will have to be encompassed in a recoverable transaction
+      async sideEffect ({ meta, contractID }, { state }) {
+        const { loggedIn } = sbp('state/vuex/state')
+        const { profiles = {} } = state
+
+        // TODO: per #257 this will ,have to be encompassed in a recoverable transaction
         // however per #610 that might be handled in handleEvent (?), or per #356 might not be needed
-        if (meta.username === rootState.loggedIn.username) {
+        if (meta.username === loggedIn.username) {
           // we're the person who just accepted the group invite
           // so subscribe to founder's IdentityContract & everyone else's
-          for (const name in state.profiles) {
-            if (name !== rootState.loggedIn.username) {
-              await sbp('chelonia/contract/sync', state.profiles[name].contractID)
+          for (const name in profiles) {
+            if (name !== loggedIn.username) {
+              await sbp('chelonia/contract/sync', profiles[name].contractID)
             }
           }
         } else {
+          const myProfile = profiles[loggedIn.username] || null
           // we're an existing member of the group getting notified that a
           // new member has joined, so subscribe to their identity contract
           await sbp('chelonia/contract/sync', meta.identityContractID)
+
+          if (!myProfile) { return }
```

Comment:
So, some consistency here is needed. Either `isActionYoungerThanUser` doesn't also do this check, in which case we should move `myProfile &&` into the `if` below, or it does do the check, in which case this line should be deleted as redundant.

Reply:
This check has been merged with the `if` below.